### PR TITLE
ceres-solver: remove gcc dependency on linux

### PR DESCRIPTION
**Pull request by fxcoudert**
As discussed in https://github.com/Homebrew/homebrew-core/pull/100411

### CODE REVIEW

File: Formula/ceres-solver.rb [hunk 1]
```diff
@@ -30,12 +29,6 @@ class CeresSolver < Formula
   depends_on "suite-sparse"
   depends_on "tbb"
 
-  on_linux do
-    depends_on "gcc"
-  end
-
-  fails_with gcc: "5" # C++17
```
Comment (carlocab) [inline Formula/ceres-solver.rb:37]: We should keep this line. The rest can be removed.

File: Formula/ceres-solver.rb [hunk 2]
```diff
@@ -18,7 +18,6 @@ class CeresSolver < Formula
     sha256 cellar: :any,                 monterey:       "ab39d83b1ae0299e88f3fc9d5b23f19c98903d6452e21b6d363966ef5f4817d0"
     sha256 cellar: :any,                 big_sur:        "d1918fe281cb00dbb2d773c03252a0d50384688c32cb1ad92510f55c37276dca"
     sha256 cellar: :any,                 catalina:       "d6a945ae9d978b449228a795cfab9ab0061920f6c73f9cb93bb0a72946f4b206"
-    sha256 cellar: :any_skip_relocation, x86_64_linux:   "5f095246dbff97818863a99864400c25278c6c23ee5bf42546eca75ae04f7de1"
```
Comment (carlocab) [inline Formula/ceres-solver.rb:21]: Let's also add a comment above the bottle block about the removal of the Linux bottle.

File: Formula/ceres-solver.rb (hunk 1 above)
Comment (fxcoudert) [reply Formula/ceres-solver.rb:37]: I'm not a linux brew user, but I don't understand what that line would achieve on its own? (And it probably also fails with some other, later gcc versions.)

Comment (carlocab) [reply Formula/ceres-solver.rb:37]: I think it should help prevent `brew` from trying to use a host `gcc-5` if that's all that's available, and we already know that that's hopeless.

**Comment by fxcoudert** [comment]
Closing in favour of the mass-PR at https://github.com/Homebrew/homebrew-core/pull/106926

**Pull request closed without merging**